File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.rs
impl Solution {
    pub fn product_except_self(nums: Vec<i32>) -> Vec<i32> {
        let (mut left, mut right) = (vec![1i32; nums.len() + 1], vec![1i32; nums.len() + 1]);
        for i in 1..=nums.len() {
            left[i] = nums[i-1]*left[i-1];
        }
        for j in (0..nums.len()).rev() {
            right[j] = nums[j]*right[j+1];
        }
        let mut res = vec![0; nums.len()];
        for k in 0..nums.len() {
            res[k] = left[k]*right[k+1];
        }
        res
    }
}

File: Data Structures & Algorithms/level-order-traversal-of-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        queue<pair<TreeNode*, int>> q;
        int last = -1;
        if (root != nullptr) {
            q.push({root, 0});
        }
        vector<vector<int>> res;
        while (!q.empty()) {
            auto p = q.front(); q.pop();
            if (last != p.second) {
                last = p.second;
                res.push_back({});
            }
            res.back().push_back(p.first->val);
            if (p.first->left != nullptr) {
                q.push({p.first->left, p.second + 1});
            }
            if (p.first->right != nullptr) {
                q.push({p.first->right, p.second + 1});
            }
        }
        return res;
    }
};
